<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="description" content="Winc FAQ">
    <meta name="author" content="Donatas Rasiukevičius">
    <title> Ik kan met promise.then fouten afvangen, maar promise.catch doet dat ook. Wat is het verschil en wanneer gebruik ik de ene versus de andere? |  Winc FAQ</title>

    <script src="https://code.jquery.com/jquery-3.5.1.js"></script>
    <script src="https://cdn.datatables.net/1.10.23/js/jquery.dataTables.min.js"></script>
    <link rel="stylesheet" href="https://cdn.datatables.net/1.10.23/css/jquery.dataTables.min.css">
    <link rel="stylesheet" href="https://static.wincacademy.nl/css/style.css">
    <link rel="stylesheet" href="/style.css">
    <link rel="icon" type="image/png" href="https://static.wincacademy.nl/logos/favicon.png">
</head>

<body>
    <div class="logo">
        <img src="https://static.wincacademy.nl/logos/main-logo.png">
    </div>
    <section>
        <header>
            <a href="/">Frequently Asked Questions</a>
        </header>
    </section>
    
<h1 id="wanneer-gebruik-ik-promise-then-versus-promise-catch">wanneer gebruik ik promise.then versus promise.catch?</h1>
<p>Als je een foutafhandelfunctie meegeeft aan .then dan vangt die alleen de error af voor die ene promise. Als je maar 1 promise gebruikt dan is dat handig.</p>
<p>Als je promises gaat chainen dan is het handig om .catch te gebruiken, die vangt dan alle errors af van alle daarvoor gechainde promises. Dan hoef je je errorafvang functie niet herhaaldelijk door te geven.</p>
<p>Zie https://github.com/WincAcademy/gists/tree/master/promises_vraag1</p>



</body>

</html>